faunadb/errors.py

```diff
@@ -1,63 +1,131 @@
 """Error types that methods in the FaunaDB client throw."""
 
-class FaunaError(Exception):
-  """Base class for all FaunaDB errors."""
-  pass
 
-class InvalidQuery(FaunaError):
-  """Thrown when a query is malformed."""
-  pass
+class InvalidResponse(Exception):
+  """Thrown when the response from the server is unusable."""
+  def __init__(self, description, response_data):
+    self.description = description
+    """Description of the kind of response expected."""
+    self.response_data = response_data
+    """Actual response data. (Type varies.)"""
+    super(InvalidResponse, self).__init__("%s\nGot: %s" % (description, response_data))
```

Comment:
I think we should leave the response out of the exception message and let them inspect the exception instead if they want it, as the response could be of an unknown size.